File: web/src/utils/systemFields.js
export const systemFields = [
    // { id: 1, name: "父工作项", systemProperty: "parent", inputType: "WORK_ITEM" },
    { id: '2', name: "工作项编号", systemProperty: "itemNo", inputType: "WORK_ITEM_NO" },
    { id: '3', name: "标题", systemProperty: "name", inputType: "TEXT" },
    { id: '4', name: "描述", systemProperty: "description", inputType: "TEXT" },
    { id: '5', name: "优先级", systemProperty: "priority", inputType: "OPTIONS" },//severity
    { id: '6', name: "负责人", systemProperty: "ownerId", inputType: "USER" },
    { id: '7', name: "创建者", systemProperty: "createBy", inputType: "USER" },
    { id: '8', name: "创建日期", systemProperty: "createDate", inputType: "DATE" },
    { id: '9', name: "修改者", systemProperty: "lastModifiedBy", inputType: "USER" },
    { id: '10', name: "修改日期", systemProperty: "lastModifiedDate", inputType: "DATE" },
    { id: '11', name: "状态", systemProperty: "status", inputType: "STATUS" },
    { id: '12', name: "状态类型", systemProperty: "meaning", inputType: "STATUS_TYPE" },
    { id: '13', name: "分配给", systemProperty: "assignedToId", inputType: "USER" },
    { id: '14', name: "分配日期", systemProperty: "assignedDate", inputType: "DATE" },
    { id: '15', name: "所属项目", systemProperty: "projectId", inputType: "PROJECT" },
    { id: '16', name: "所属工作项类型", systemProperty: "trackerId", inputType: "WORK_ITEM_TYPE" },
    { id: '27', name: "所属迭代", systemProperty: "sprintId", inputType: "SPRINT" },
    { id: '18', name: "计划开始时间", systemProperty: "planStartDate", inputType: "DATE" },
    { id: '19', name: "计划结束时间", systemProperty: "planEndDate", inputType: "DATE" },
    { id: '20', name: "实际开始时间", systemProperty: "realStartDate", inputType: "DATE" },
    { id: '21', name: "实际结束时间", systemProperty: "realEndDate", inputType: "DATE" },
    { id: '22', name: "关闭时间", systemProperty: "closeDate", inputType: "DATE" },
    { id: '23', name: "预计花费工时", systemProperty: "estimateWorkingHours", inputType: "INTEGER" },
    { id: '24', name: "已登记工时", systemProperty: "registeredWorkingHours", inputType: "INTEGER" },
    { id: '25', name: "剩余工时", systemProperty: "remainingWorkingHours", inputType: "INTEGER" },
    { id: '26', name: "关注者", systemProperty: "watchers", inputType: "USER" },
]

File: server/helm-tracker/src/main/java/com/harbortek/helm/tracker/constants/SystemFields.java
/*
 * Copyright [2025] [Harbortek Corp.]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.harbortek.helm.tracker.constants;

import java.util.Arrays;
import java.util.List;

public interface SystemFields {

    String PARENT = "parent";
    String ITEM_NO = "itemNo";

    String NAME = "name";

    String DESCRIPTION = "description";

    String OWNER = "ownerId";
    String SPRINT = "sprintId";

    String CREATE_BY = "createBy";

    String CREATE_DATE = "createDate";


    String LAST_MODIFIED_BY = "lastModifiedBy";

    String LAST_MODIFIED_DATE = "lastModifiedDate";

    String STATUS = "status";

    String STATUS_TYPE = "meaning";

    String ASSIGNED_TO = "assignedToId";

    String ASSIGNED_DATE = "assignedDate";

    String PROJECT = "projectId";

    String TRACKER = "trackerId";

    String PRIORITY = "priority";

    /**
     * 计划开始时间
     */
    String PLAN_START_DATE = "planStartDate";

    /**
     * 计划结束时间
     */
    String PLAN_END_DATE = "planEndDate";

    /**
     * 实际开始时间
     */
    String REAL_START_DATE = "realStartDate";
    /**
     * 实际结束时间
     */
    String REAL_END_DATE = "realEndDate";

    /**
     * 进度
     */
    String PROGRESS = "progress";

    /**
     * 关闭时间
     */
    String CLOSE_DATE = "closeDate";

    /**
     * 预计花费工时
     */
    String ESTIMATE_WORKING_HOURS = "estimateWorkingHours";
    /**
     * 已登记工时
     */
    String REGISTERED_WORKING_HOURS = "registeredWorkingHours";

    /**
     * 剩余工时
     */
    String REMAINING_WORKING_HOURS = "remainingWorkingHours";

    /**
     * 测试用例类型
     */
//    String TEST_CASE_TYPE = "testCaseTypeId";
//
//    String PRECONDITION = "precondition";

    String SEVERITY = "severity";
    String DUE_DATE = "dueDate";

    String RESOLUTION = "resolution";

    String RESOLVED_ON = "resolvedOn";


    List<String> ALL_SYSTEM_FIELDS = Arrays.asList(
//            PARENT,
            ITEM_NO,
            NAME,
            DESCRIPTION,
            OWNER,
            SPRINT,
            CREATE_BY,
            CREATE_DATE,
            LAST_MODIFIED_BY,
            LAST_MODIFIED_DATE,
            STATUS,
            STATUS_TYPE,
            ASSIGNED_TO,
            ASSIGNED_DATE,
            SEVERITY,
            DUE_DATE,
            PROJECT,
            TRACKER,
            PRIORITY,
            SEVERITY,
            PLAN_START_DATE,
            PLAN_END_DATE,
            REAL_START_DATE,
            REAL_END_DATE,
            PROGRESS,
            CLOSE_DATE,
            RESOLUTION,
            RESOLVED_ON,
            ESTIMATE_WORKING_HOURS,
            REGISTERED_WORKING_HOURS,
            REMAINING_WORKING_HOURS
    );


}
